plugins/Files/js/file.js

```diff
@@ -98,42 +108,47 @@ var file = {
 		// path of 'foo/bar/baz' would return ['foo', 'bar']
 		return this.directory.split('/');
 	},
-	get parentFolders () {
-		// path of 'foo/bar/baz' would return folder objects whose paths are
-		// ['foo', 'foo/bar'] respectively
-		var parentFolders = [];
-		// iterate through parentFolder links to populate parentFolders
-		for (let i = this.parentFolder; i; i = i.parentFolder) {
-			parentFolders.push(i);
-		}
-		return parentFolders.reverse();
-	},
 	get extension () {
 		return path.extname(this.path);
 	},
 	get nameNoExtension () {
 		return path.basename(this.path, this.extension);
 	},
-	get size () {
-		return this.filesize;
-	},
-	get count () {
-		return 1;
-	},
 
 	// These can't use set syntax because they're necessarily asynchronous
 	setName (newName, cb) {
+		if (tools.notType(newName, 'string')) {
+			return;
+		}
 		var newPath = `${this.directory}/${newName}`;
 		this.setPath(newPath, cb);
 	},
 	setDirectory (newDirectory, cb) {
+		if (tools.notType(newDirectory, 'string')) {
+			return;
+		}
 		var newPath = `${newDirectory}/${this.name}`;
 		this.setPath(newPath, cb);
 	},
 	setExtension (newExtension, cb) {
+		if (tools.notType(newExtension, 'string')) {
+			return;
+		}
 		var newPath = `${this.directory}/${this.nameNoExtension}${newExtension}`;
 		this.setPath(newPath, cb);
 	},
+
+	// Misc. functions
+	get parentFolders () {
+		// path of 'foo/bar/baz' would return folder objects whose paths are
+		// ['foo', 'foo/bar'] respectively
+		var parentFolders = [];
+		// iterate through parentFolder links to populate parentFolders
+		for (let i = this.parentFolder; i; i = i.parentFolder) {
+			parentFolders.push(i);
+		}
+		return parentFolders.reverse();
```

Comment:
This gets references to the folder objects, not the array of folder names (as the function right above it does)